### PR TITLE
fix: Correctly treat `NotFound` in terms of cache fallback

### PR DESCRIPTION
**Pull request by Swatinem**
We have a steady stream of cache fallbacks since a while which has been baffling me.

I suspect these happen because:
- we have an older cache version to use
- when regenerating the cache, the file can not be found anymore and a `NotFound` marker is written
- the `NotFound` marker was previously always skipped, falling back to the old cache version

Apart from fixing and testing this edge case, this also simplifies the cache versioning code a bit.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/cacher.rs [hunk 1]
```diff
@@ -694,6 +682,42 @@ mod tests {
...
+        test::setup();
+
+        let cache_dir = test::tempdir().path().join("test");
+        std::fs::create_dir_all(cache_dir.join("global/1")).unwrap();
+        std::fs::write(
+            cache_dir.join("global/some_cache_key"),
+            "some old cached contents",
+        )
+        .unwrap();
+        std::fs::create_dir_all(cache_dir.join("1/global")).unwrap();
```
Comment (loewenheim) [inline crates/symbolicator-service/src/services/cacher.rs:697]: Is there intentionally both `global/1` and `1/global`?

Comment (Swatinem) [reply crates/symbolicator-service/src/services/cacher.rs:697]: good catch! I got confused first until I remembered that the cache key is `$version/$scope` and not the other way around :-)

**Review by loewenheim — APPROVED** [review]
